cf/field.py

```diff
@@ -12458,104 +12458,94 @@ def flip(self, axes=None, inplace=False, i=False, **kwargs):
 
         return f
 
-    def argmax(self, axis=None):
+    def argmax(self, axis=None, unravel=False):
         """Return the indices of the maximum values along an axis.
 
         If no axis is specified then the returned index locates the
         maximum of the whole data.
 
-        .. seealso:: `argmin`, `where`
+        In case of multiple occurrences of the maximum values, the
+        indices corresponding to the first occurrence are returned.
+
+        **Performance**
+
+        If the data index is returned as a `tuple` (see the *unravel*
+        parameter) then all delayed operations are computed.
+
+        .. seealso:: `where`, `cf.Data.argmax`
 
         :Parameters:
 
+            axis: optional
+                Select the domain axis over which to locate the
+                maximum values, defined by the domain axis that would
+                be selected by passing the given *ais* to a call of
+                the field construct's `domain_axis` method. For
+                example, for a value of ``'X'``, the domain axis
+                construct returned by ``f.domain_axis('X')`` is
+                selected.
+
+                By default the maximum over the flattened data is
+                located.
+
+            unravel: `bool`, optional
+                If True then when locating the maximum over the whole
+                data, return the location as an index for each axis as
+                a `tuple`. By default an index to the flattened array
+                is returned in this case. Ignored if locating the
+                maxima over a subset of the axes.
+
         :Returns:
 
+            `Data` or `tuple`
```

Comment:
```suggestion
            `Data` or `tuple` of `Data`
```

to be specific?

Reply:
It's OK, but it should by tuple of `int`: https://github.com/NCAS-CMS/cf-python/pull/557/commits/f8e3b407ff4e60c4e688dccef47ba848a28522a2